app/controllers/user_mentee_applications_controller.rb

```diff
@@ -17,23 +18,27 @@ def new
     end
   end
 
+  # rubocop:disable Metrics/AbcSize
   def create
     if current_user.mentee_application.present?
       redirect_to user_mentee_application_path(current_user.mentee_application)
     end
 
-    @user_mentee_application = authorize UserMenteeApplication.new(user_mentee_application_params)
-    @user_mentee_application.user = current_user
-    @user_mentee_application.user_mentee_application_cohort = UserMenteeApplicationCohort.active
-
-    respond_to do |format|
-      if @user_mentee_application.save
-        format.html { redirect_to @user_mentee_application, notice: 'Application succesfully submitted' }
-      else
-        format.html { render :new, status: :unprocessable_entity }
-      end
+    @user_mentee_application = authorize UserMenteeApplication.new(
+      user: current_user,
+      user_mentee_application_cohort: UserMenteeApplicationCohort.active,
+      **user_mentee_application_params
+    )
+    ActiveRecord::Base.transaction do
+      @user_mentee_application.save!
+      current_user.resumes.create!(resume: resume_params[:resume], name: resume_params[:resume_name], current: true)
```

Comment:
Can we use the Resumes::UpdateService here? Especially because they might have an existing resume if they reapply.

Reply:
I chose not to leverage it in because it seemed like the likelihood of someone selecting a previous uploaded resume was low, especially considering that, as it stands, once they submit, that's their submission. How will the reapplication flow look like? I think maybe we can keep it out until that flow is better fleshed out?

Reply:
I'll add that the service requires passing in a current resume id param, too. Right now, that doesn't seem like something that's needed on this form. Given that, I figure it's simpler to keep it like this.

Reply:
ok sounds good.